emote/nn/layers.py

```diff
@@ -78,6 +84,95 @@ def get_encoder_output_size(self, flatten: bool = False):
             curr_size_y = ((curr_size_y - k + 2 * p) // s) + 1
 
         out_size = (self._channels[-1], curr_size_x, curr_size_y)
-        if flatten:
-            out_size = np.prod(out_size)
+
+        if self.flatten or return_flattened:
+            flattened_out_size = np.prod(out_size)
+            return *out_size, flattened_out_size
```

Comment:
Sorry if I might have missed this, but why do we return this thing and not just `flattened_out_size`? It doesn't really make sense to me, and if we do want to return both the unflattened sizes and the flattened one, why expand the first tuple? Wouldn't it be easier to read if we just returned `(out_size, flattened_out_size)`?

Reply:
I added the unflattened one for logging purposes, but I think you are right that it doesn't make much sense to keep it. I will remove it and update the code :)